devops/DeployCloudformationS3Bucket/deployStackS3Bucket.py

```diff
@@ -1,32 +1,54 @@
-#Deploys a Stack and S3 Bucket To Cloudformation.
+#Deploys a stack and S3bucket to cloudformation.
+
+#Imports Python libraries
 import boto3
 import re
 import os
 import sys
 import argparse
-
+import random
+import string
+import subprocess
+from datetime import date
 from os.path import dirname
 script_dir = dirname(__file__)
 
-#Creates Cloudformation Client
-cf_client = boto3.client('cloudformation')
+#Creates cloudformation client
+cf_client = boto3.client('cloudformation', region_name='us-west-2')
 
+#Requires user initials input parameter.
 def main(input_initials):
 
-    #Read In Stack Template File
+    #Read in stack template file
     stack_template = _stack_template_file()
 
-    #Parses Input Initials
+    #Parses input initials
     parser = argparse.ArgumentParser(description='Input Initials Required.')
-    parser.add_argument('input_initials', metavar='input_initials', help='Input User Initials For Stack Name.', nargs='?', type=str)
+    parser.add_argument('input_initials', metavar='input_initials', help='Input User Initials To Use In Stack Name.', nargs='?', type=str)
     args = parser.parse_args()
 
-    #Gets Initials
+    #Gets initials
     initials = args.input_initials
     print("\nYour Initials Entered: " + initials + "\n")
 
-    #Uses Initials For Stack Name
-    stack_name = (initials + '-stack')
+    #Gererates unique id
+    unique_id = ''.join(random.SystemRandom().choice(string.ascii_lowercase + string.digits) for _ in range(5))
+
+    #Gets the date
+    today = date.today()
+    date_created = today.strftime("%m%d%y")
+
+    #Uses initials, date, and unique id for stack name
+    delimiter = "-"
+    stack_vars = (initials, date_created, unique_id, 'stack')
+    stack_name = delimiter.join(stack_vars)
+    
+    #Uses stack with bucket name
+    bucket_vars = (stack_name, 'reactapp', 'bucket')
+    bucket_name = delimiter.join(bucket_vars)
+
+    #Creates bucket parameters
+    parameters=[ { 'ParameterKey': 'BucketName', 'ParameterValue': bucket_name } ]
     
     #Verifies Initials
     patternAlpha = re.compile("[A-Za-z]+")
```

Comment:
Recommend making this into a function and removing from your main program path

Reply:
@eknigge I feel that the override bucket name parameter location makes sense in this case, but thank you for the suggestion.👍

Reply:
Sorry for not being clear the first time. I mean lines 57-84 could be abstracted way by using a function. The rest of the code is easy to follow, but it clutters your main method to include all of the core logic there.

Reply:
@eknigge I moved the setting parameters and logic to separate functions and added logging instead of using print method.
If you run the  deployStackS3Bucket.py script on local cmd, a log file deployStackS3Bucket.log will be created and logs will be sent to file.